src/pool/mempool.rs

```diff
@@ -38,31 +37,25 @@ pub type TransactionOrdering = CoinbaseTipOrdering<EthPooledTransaction>;
 /// A type alias for the Kakarot Sequencer Mempool.
 pub type KakarotPool<Client> = Pool<Validator<Client>, TransactionOrdering, NoopBlobStore>;
 
-/// Manages a collection of accounts and their associated nonces, interfacing with an Ethereum client.
+/// Manages a collection of accounts addresses, interfacing with an Ethereum client.
 ///
 /// This struct provides functionality to initialize account data from a file, monitor account balances,
 /// and process transactions for accounts with sufficient balance.
 #[derive(Debug)]
 pub struct AccountManager<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> {
-    /// A shared, mutable collection of accounts and their nonces.
-    accounts: HashMap<Felt, Arc<Mutex<Felt>>>,
+    /// A collection of account addresses.
+    accounts: Vec<Felt>,
     /// The Ethereum client used to interact with the blockchain.
     eth_client: Arc<EthClient<SP>>,
 }
 
 impl<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> AccountManager<SP> {
     /// Initialize the account manager with a set of passed accounts.
-    pub async fn from_addresses(addresses: Vec<Felt>, eth_client: Arc<EthClient<SP>>) -> eyre::Result<Self> {
-        let mut accounts = HashMap::new();
+    pub fn from_addresses(addresses: Vec<Felt>, eth_client: Arc<EthClient<SP>>) -> eyre::Result<Self> {
+        let mut accounts = Vec::new();
 
         for add in addresses {
-            // Query the initial account_nonce for the account from the provider
-            let nonce = eth_client
-                .starknet_provider()
-                .get_nonce(starknet::core::types::BlockId::Tag(BlockTag::Pending), add)
-                .await
-                .unwrap_or_default();
-            accounts.insert(add, Arc::new(Mutex::new(nonce)));
+            accounts.push(add);
         }
```

Comment:
this can be `new` now no? just pass the addresses and client in and do `Self{addresses, eth_client}`

Reply:
Yes good point, now this is straightforward to have a `new`, just fixed